raster-foundry/raster-foundry
# Add frontend and API for setting Planet token

## Overview

Add frontend and api for setting planet api token

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo
![image](https://user-images.githubusercontent.com/4392704/28925336-9a4e5f7e-7832-11e7-888a-2d94d37c7733.png)

![image](https://user-images.githubusercontent.com/4392704/28925343-a19c75ea-7832-11e7-9890-3f32ce1a7fd8.png)

### Notes
See #2372 
This api will work for now, but I we need to handle how we structure the `/users/me` since right now it's used to get the combined auth0User object, patch it, and update the token. We can support doing all that in one place, but we need a better structure to how we represent the user since at this point we have 3 formats - auth0 user token, combined user & auth0 user, and our database user.

## Testing Instructions

* Verify that making a PUT request to the `/api/users/me` endpoint with with your user object is only able to change the modified time and planet token
* Verify that frontend acknowledges a successfully set token by changing the button text

Closes #2339
